ledger/accountdb_test.go

```diff
@@ -2806,3 +2807,125 @@ func TestAccountsNewRoundDeletedResourceEntries(t *testing.T) {
+	dbs, fn := dbOpenTest(b, false)
+	setDbLogging(b, dbs)
+	cleanup := func() {
+		cleanupTestDb(dbs, fn, false)
+	}
+
+	tx, err := dbs.Wdb.Handle.Begin()
+	require.NoError(b, err)
+	_, err = accountsInit(tx, make(map[basics.Address]basics.AccountData), proto)
+	require.NoError(b, err)
+	err = tx.Commit()
+	require.NoError(b, err)
+	err = dbs.Wdb.SetSynchronousMode(context.Background(), db.SynchronousModeOff, false)
+	require.NoError(b, err)
+
+	cnt := 0
+	for batch := 0; batch <= batchCount; batch++ {
+		tx, err = dbs.Wdb.Handle.Begin()
+		require.NoError(b, err)
+		writer, err := makeAccountsSQLWriter(tx, false, false, true, false)
+		require.NoError(b, err)
+		for boxIdx := 0; boxIdx < totalBoxes/batchCount; boxIdx++ {
+			err = writer.upsertKvPair(fmt.Sprintf("%d", cnt), string(make([]byte, boxSize)))
+			require.NoError(b, err)
+			cnt++
+		}
+
+		err = tx.Commit()
+		require.NoError(b, err)
+		writer.close()
+	}
+	err = dbs.Wdb.SetSynchronousMode(context.Background(), db.SynchronousModeFull, true)
+	return dbs, cleanup, err
+}
+
+func BenchmarkBoxDatabaseRead(b *testing.B) {
+	getBoxNamePermutation := func(totalBoxes int) []int {
+		rand.Seed(time.Now().UnixNano())
+		boxNames := make([]int, totalBoxes)
+		for i := 0; i < totalBoxes; i++ {
+			boxNames[i] = i
+		}
+		rand.Shuffle(len(boxNames), func(x, y int) { boxNames[x], boxNames[y] = boxNames[y], boxNames[x] })
+		return boxNames
+	}
+
+	boxCnt := []int{10, 1000, 100000}
+	boxSizes := []int{2, 2048, 4 * 8096}
+	for _, totalBoxes := range boxCnt {
+		for _, boxSize := range boxSizes {
+			b.Run(fmt.Sprintf("totalBoxes=%d/boxSize=%d", totalBoxes, boxSize), func(b *testing.B) {
+				b.StopTimer()
+
+				dbs, cleanup, err := initBoxDatabase(b, totalBoxes, boxSize)
+				require.NoError(b, err)
+
+				boxNames := getBoxNamePermutation(totalBoxes)
+				var v sql.NullString
+				for i := 0; i < b.N; i++ {
+					lookupStmt, err := dbs.Wdb.Handle.Prepare("SELECT rnd, value FROM acctrounds LEFT JOIN kvstore ON key = ? WHERE id='acctbase';")
+					require.NoError(b, err)
+
+					var pv persistedValue
+					boxName := boxNames[i%totalBoxes]
+					b.StartTimer()
+					err = lookupStmt.QueryRow([]byte(fmt.Sprintf("%d", boxName))).Scan(&pv.round, &v)
+					b.StopTimer()
+					require.NoError(b, err)
+					require.True(b, v.Valid)
+				}
+
+				cleanup()
+			})
+		}
+	}
+
+	lookbacks := []int{2, 32, 256, 2048}
+	for _, lookback := range lookbacks {
+		for _, boxSize := range boxSizes {
+			totalBoxes := 100000
+
+			b.Run(fmt.Sprintf("lookback=%d/boxSize=%d", lookback, boxSize), func(b *testing.B) {
+				b.StopTimer()
+
+				dbs, cleanup, err := initBoxDatabase(b, totalBoxes, boxSize)
+				require.NoError(b, err)
+
+				boxNames := getBoxNamePermutation(totalBoxes)
+				var v sql.NullString
+				for i := 0; i < b.N+lookback; i++ {
+					lookupStmt, err := dbs.Wdb.Handle.Prepare("SELECT rnd, value FROM acctrounds LEFT JOIN kvstore ON key = ? WHERE id='acctbase';")
```

Comment:
@algoidurovic For my understanding:  Is it a by-design move to create a new `Stmt` for each look up?

Here's some notes + thoughts.  I can discuss live if easier.
* I haven't spent much time with `Prepare` though I imagine `Stmt` _can_ be reused.    And it seems transaction processing always _reuses_ a single `Stmt`.
* Of note, the test's configured behavior is to always open a new connection.  And from my brief inspection, no max connection limit is configured.  
* I don't have a feel for test variance yet though running with reused `Stmt` does change results.  Generally, I'm concerned always creating new connections varies from transaction processing behavior + _may_ artificially slow read performance.